### PR TITLE
Merge dev into main (release 0.21.2)

### PR DESCRIPTION
**Pull request by Pronyo-Chan**

### CODE REVIEW

**Comment by icebound777** [comment]
PR title says 0.21.3, discord comment and bps files say 0.21.2?

**Comment by Pronyo-Chan** [comment]
Counting is hard, sorry about the wrong version in title, fixed it

File: rando_modules/logic.py [hunk 1]
```diff
@@ -821,7 +822,6 @@ def _algo_assumed_fill(
     print("Placing progression items...")
     #Place progression items, both key and replenishable
     pool_combined_progression_items = pool_progression_items + pool_misc_progression_items
-    random.shuffle(pool_combined_progression_items)
```
Comment (icebound777) [inline rando_modules/logic.py:824]: This turns of shuffling the item pool, was this intended?

Comment (Pronyo-Chan) [reply rando_modules/logic.py]: Nope, big blunder, unnoticed local edit that got commited by mistake. Reverted.

**Comment by icebound777** [comment]
Assuming the bps is of the correct base mod branch: lgtm

**Comment by Pronyo-Chan** [comment]
lgtm

**Pull request merged**